File: backend/src/main/java/com/tododuk/domain/reminder/dto/ReminderDto.java
package com.tododuk.domain.reminder.dto;

import com.tododuk.domain.reminder.entity.Reminder;
import com.tododuk.domain.todo.entity.Todo;
import lombok.NonNull;

import java.time.LocalDateTime;

public record ReminderDto(
        @NonNull Todo todo,
        @NonNull LocalDateTime remindAt,
        @NonNull String method,
        @NonNull int id,
        @NonNull LocalDateTime createDate,
        @NonNull LocalDateTime modifyDate
){

    public ReminderDto(Reminder reminder) {
        this(
                reminder.getTodo(),
                reminder.getRemindAt(),
                reminder.getMethod(),
                reminder.getId(),
                reminder.getCreateDate(),
                reminder.getModifyDate()
        );
    }
}

File: backend/src/main/java/com/tododuk/domain/reminder/dto/ReminderDto.kt
package com.tododuk.domain.reminder.dto

import com.tododuk.domain.reminder.entity.Reminder
import java.time.LocalDateTime

@JvmRecord
data class ReminderDto(
    val todoId: Int?,
    val remindAt: LocalDateTime,
    val method: String,
    val id: Int,
    val createDate: LocalDateTime,
    val modifyDate: LocalDateTime
) {
    constructor(reminder: Reminder) : this(
        reminder.todoId,
        reminder.remindAt,
        reminder.method,
        reminder.id,
        reminder.createDate,
        reminder.modifyDate
    )
}